src/dp_abstraction/aca_goal_state_handler.cpp

```diff
@@ -329,14 +364,28 @@ int Aca_Goal_State_Handler::update_neighbor_state_workitem_v2(
 int Aca_Goal_State_Handler::update_neighbor_states(GoalStateV2 &parsed_struct,
                                                    GoalStateOperationReply &gsOperationReply)
 {
+  auto t = std::chrono::high_resolution_clock::now();
+  std::cout<< "Start processing neighbor states at "<<t.time_since_epoch().count()<<std::endl;
   std::vector<std::future<int> > workitem_future;
   int rc;
   int overall_rc = EXIT_SUCCESS;
   int count = 1;
-
+  GoalStateV2* gsv2_ptr = &parsed_struct;
+  GoalStateOperationReply* reply_ptr = &gsOperationReply;
+  marl::WaitGroup wait_group(parsed_struct.neighbor_states_size());
+  
   for (auto &[neighbor_id, current_NeighborState] : parsed_struct.neighbor_states()) {
-    ACA_LOG_DEBUG("=====>parsing neighbor state: %s\n", neighbor_id.c_str());
-
+    //ACA_LOG_DEBUG("=====>parsing neighbor state: %s\n", neighbor_id.c_str());
+    // neighbor_count.fetch_add(1);
+    // neighbor_wait_group.add();
+    packet_in_counter++;
+    marl::schedule([=] {
+      defer(wait_group.done());
+      update_neighbor_state_workitem_v2(current_NeighborState, *gsv2_ptr, *reply_ptr);
+    });
+  }
+  wait_group.wait();
```

Comment:
remove unstable  wait_group

Reply:
We shall keep the waitgroup as we need to wait for all states to be finished until going into the next step.